juju/juju
# feat(dqlite): update workers to use new WatchMachines

> [!NOTE]
> This was growing too much, so work is split out to https://github.com/juju/juju/pull/17558. This PR will be rebased onto that. The rest of description is kept for now to be used during the rebase and QA.

This adds the following:

- Updates `ModelMachinesWatcher` on the apiserver to use the new `WatchModelMachines` on the machine domain service.
- Updates the `provisioner`, `firewaller` and the `instancepoller` facades to pass the machine service to the ModelMachinesWorker.

## Checklist




> [!CAUTION]
> Currently for a reason I've yet to understand, this causes firewaller to bork after bootstrap with:
``` $ jdebug -m controller
unit-controller-0: 14:52:33 INFO juju.worker.uniter.operation ran "update-status" hook (via hook dispatching script: dispatch)
controller-0: 14:52:41 INFO juju.worker.firewaller runner is dying
machine-0: 14:52:41 ERROR juju.worker.dependency "firewaller" manifold worker returned unexpected error: permission denied 
```

## QA steps

All following should work on lxd and aws.

```
 $ juju bootstrap lxd
 $ juju add-model deleteme
 $ juju deploy ubuntu
 $ juju add-machine -n 2
 $ juju add-machine lxd:1
```

Also some unit tests of interest:

```
 $ TEST_PACKAGES="./domain/schema/... -count=1 -race" make run-go-tests
 $ TEST_PACKAGES="./domain/machine/... -count=1 -race" make run-go-tests
 $ TEST_PACKAGES="./apiserver/facades/controller/firewaller/... -count=1 -race" make run-go-tests
 $ TEST_PACKAGES="./apiserver/facades/controller/instancepoller/... -count=1 -race" make run-go-tests
```

## Links

**Jira card:** JUJU-
